src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileApp.cs

```diff
@@ -633,24 +633,33 @@ public void ILLink_analyzer_warnings_are_produced_using_EnableSingleFileAnalyzer
         }
 
         [RequiresMSBuildVersionTheory("17.0.0.32901")]
-        [InlineData("netstandard2.0")]
-        public void EnableSingleFileAnalyzer_warns_for_unsupported_target_framework(string targetFramework)
+        [InlineData("netstandard2.0", true)]
+        [InlineData("netstandard2.0;net5.0", true)] // None of these TFMs are supported for single-file
+        [InlineData("netstandard2.0;net6.0", false)] // Net6.0 is the min TFM supported for single-file and targeting.
+        [InlineData("netstandard2.0;net8.0", true)] // Net8.0 is supported for single-file, but leaves a "gap" for the supported net6./net7.0 TFMs.
+        public void EnableSingleFile_warns_when_expected_for_not_correctly_multitargeted_libraries(string targetFrameworks, bool shouldWarn)
         {
+            
             var testProject = new TestProject()
             {
                 Name = "ClassLibTest",
-                TargetFrameworks = targetFramework
+                TargetFrameworks = targetFrameworks
```

Comment:
Do we have a test that explicitly sets `<TargetFramework>netstandard2.0</TargetFramework>`? Note that is using the singular TargetFramework and not the plural TargetFrameworks.

Reply:
The `TestProject` class being used here conflates the two, ie it only offers the plural `TargetFrameworks` property, and then if there's no semicolon in it, it will write the value to the `TargetFramework` property in the project instead.